include/ndarray.hpp

```diff
@@ -110,20 +117,24 @@ class NDArray {
   bool c_continuous() const;
 
   // Save array to the file fname.npy
-  void save(std::string fname) const;
+  void save(const std::string& fname) const;
 
   //==========================================================================
   // Non-Constant Methods
 
   // Fills entire array with the value provided
-  void fill(T val);
+  void fill(const T& val);
 
   // Will reshape the array to the given dimensions
-  void reshape(std::vector<size_t> new_shape);
+  void reshape(const std::vector<size_t>& new_shape);
 
   // Realocates array to fit the new size.
   // DATA CAN BE LOST IF ARRAY IS SHRUNK
-  void reallocate(std::vector<size_t> new_shape);
+  void reallocate(const std::vector<size_t>& new_shape);
+
+  // Assignment Operator
+  NDArray& operator=(const NDArray&) = default;
```

Comment:
These are fine, but could they just be moved to the top of the class, so they are with the constructors ?

Reply:
Your comment is just right. I have moved the declaration of assignment operators near constructors with former commit aef7e48 and updated commit b7886e2.
diff: https://github.com/munehiro-k/ndarray/compare/aef7e48..b7886e2